cmd/lakefs/cmd/run.go

```diff
@@ -174,12 +176,18 @@ var runCmd = &cobra.Command{
 		logging.Default().WithField("listen_address", cfg.GetListenAddress()).Info("starting HTTP server")
 		server := &http.Server{
 			Addr: cfg.GetListenAddress(),
-			Handler: httputil.HostMux(
-				httputil.HostHandler(apiHandler).Default(), // api as default handler
-				httputil.HostHandler(s3gatewayHandler, // s3 gateway for its bare domain and sub-domains of that
-					httputil.Exact(cfg.GetS3GatewayDomainNames()),
-					httputil.SubdomainsOf(cfg.GetS3GatewayDomainNames())),
-			),
+			Handler: http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+				// If the request has the S3 GW domain (exact or subdomain) - or carries an AWS sig, serve S3GW
+				if httputil.HostMatches(request, cfg.GetS3GatewayDomainNames()) ||
```

Comment:
Looks like you switched from using the generic solution called HostMux - can we throw some code?

Reply:
Already thrown :)